### ISSUE TITLE
Speed up reads by adding support for lazy decoding of registers

### ISSUE THREAD

**Issue by fxamacker** [onflow/atree#341]
### Issue To Be Solved

Decoding should be optimized to improve speed.  Currently, Atree encodes in parallel (multiple goroutines) but decodes sequentially.  Additionally, the average size of registers will increase when Atree Inlining is deployed.

### Suggested Solution

Speed up reads by adding support for lazy decoding of registers.

**Comment by fxamacker** [comment]
This was postponed so I can work on [atree v0.8.0](https://github.com/onflow/atree/releases/tag/v0.8.0) (e.g. atree inlining & deduplication, etc.) and also help at `onflow/cadence` and `onflow/flow-go` repos with integrations, migrations, tooling, and testing related to atree, etc.

The speed optimizations in this issue would not require spork or migration.

**Comment by fxamacker** [comment]
This is blocked by
- #464

cc: @j1010001

**Comment by fxamacker** [comment]
I was asked not to begin work on this several weeks ago because Atree is "efficient enough" (based on profiling report).
